### DISCUSSION TITLE
STM32?

### DISCUSSION THREAD

**Discussion by victorallume** [spatialdude/usqlite#16] [Q&A] [answered]
I'm looking at running this on an STM32 device, but it looks like that port isn't built using cmake. Is there anything you can think of that would prevent it from running on stm32? What's required to build using vanilla make?

**Comment by spatialdude** [comment, marked as answer]
@victorallume Compiling for an STM32 would be much like other platforms.

e.g. https://github.com/spatialdude/usqlite/wiki/unix but with `cd ~/myproject/micropython/ports/stm32` instead.

Although I don't have a STM32 device, I did a quick test and it appears there are some inital errors due to the size of a float on the STM32 - `error: floating constant exceeds range of 'float' [-Werror=overflow]`.

To work through these issues you can look at modifying the [default SQLite configuration options](https://github.com/spatialdude/usqlite#custom-configurations), also refer to the [SQLite compile-time options documentation](https://sqlite.org/compile.html).

If you wish to disable certain compiler errors then modify [`usqlite.c`](https://github.com/spatialdude/usqlite/blob/0fc1f3b1d0dcb4d13c41f148c00b30639717d5ed/usqlite.c#L32) accordingly.
